api/transactions/transactions_benchmark_test.go

```diff
@@ -0,0 +1,536 @@
+	newChain, transactions := createPackedChain(b, muxdb.NewMem(), blockCount, accounts, randomSignerFunc)
+
+	// shuffle the transaction into a randomized order
+	randomizedTransactions := shuffleSlice(transactions)
+	b.Logf("About to process %d txs", len(randomizedTransactions))
+
+	// run the benchmarks
+	b.Run("getTransaction", func(b *testing.B) {
+		benchmarkGetTransaction(b, newChain, randomizedTransactions)
+	})
+
+	b.Run("getReceipt", func(b *testing.B) {
+		benchmarkGetReceipt(b, newChain, randomizedTransactions)
+	})
+}
+
+func benchmarkGetTransaction(b *testing.B, thorChain *testchain.Chain, randTxs tx.Transactions) {
+	mempool := txpool.New(thorChain.Repo(), thorChain.Stater(), txpool.Options{Limit: 10, LimitPerAccount: 16, MaxLifetime: 10 * time.Minute})
+	transactionAPI := New(thorChain.Repo(), mempool)
+	head := thorChain.Repo().BestBlockSummary().Header.ID()
+	var err error
+
+	// Measure memory usage
+	b.ReportAllocs()
+
+	// Benchmark execution
+	b.ResetTimer()
+
+	for _, randTx := range randTxs {
+		_, err = transactionAPI.getRawTransaction(randTx.ID(), head, false)
+		if err != nil {
+			b.Fatalf("getRawTransaction failed: %v", err)
+		}
+	}
+}
+
+func benchmarkGetReceipt(b *testing.B, thorChain *testchain.Chain, randTxs tx.Transactions) {
+	mempool := txpool.New(thorChain.Repo(), thorChain.Stater(), txpool.Options{Limit: 10, LimitPerAccount: 16, MaxLifetime: 10 * time.Minute})
+	transactionAPI := New(thorChain.Repo(), mempool)
+	head := thorChain.Repo().BestBlockSummary().Header.ID()
+	var err error
+
+	// Measure memory usage
+	b.ReportAllocs()
+
+	// Benchmark execution
+	b.ResetTimer()
+
+	for _, randTx := range randTxs {
+		_, err = transactionAPI.getTransactionReceiptByID(randTx.ID(), head)
+		if err != nil {
+			b.Fatalf("getTransactionReceiptByID failed: %v", err)
+		}
+	}
+}
+
+func createPackedChain(b *testing.B, db *muxdb.MuxDB, noBlocks int, accounts []genesis.DevAccount, createTxFunc func(chain *testchain.Chain) (tx.Transactions, error)) (*testchain.Chain, tx.Transactions) {
+	proposer := &accounts[0]
+
+	// mock a fake chain for block production
+	fakeChain, err := createChain(db, accounts)
+	require.NoError(b, err)
+
+	// pre-alloc blocks
+	var transactions tx.Transactions
+
+	// Start from the Genesis block
+	previousBlock := fakeChain.GenesisBlock()
+	for i := 0; i < noBlocks; i++ {
+		newTxs, err := createTxFunc(fakeChain)
+		require.NoError(b, err)
+		previousBlock, err = packTxsIntoBlock(
+			fakeChain,
+			proposer,
+			previousBlock,
+			newTxs,
+		)
+		require.NoError(b, err)
+		transactions = append(transactions, newTxs...)
+	}
+
+	return fakeChain, transactions
+}
+
+func createOneClausePerTx(signerPK *ecdsa.PrivateKey, thorChain *testchain.Chain) (tx.Transactions, error) {
+	var transactions tx.Transactions
+	gasUsed := uint64(0)
+	for gasUsed < 9_500_000 {
+		toAddr := datagen.RandAddress()
+		cla := tx.NewClause(&toAddr).WithValue(big.NewInt(10000))
+		transaction := new(tx.LegacyBuilder).
```

Comment:
I guess we could have either mixed or test for each scenario (legacy, dyn and mixed)

Reply:
Will be added in the next iteration